ddtrace/mocktracer/mocktracer_test.go

```diff
@@ -92,21 +94,44 @@ func TestTracerFinishedSpans(t *testing.T) {
 	assert.Equal(t, 2, found)
 }
 
-func TestTracerReset(t *testing.T) {
-	var mt mocktracer
-	mt.StartSpan("db.query").Finish()
+func TestTracerOpenSpans(t *testing.T) {
+	mt := newMockTracer()
+	assert.Empty(t, mt.OpenSpans())
+	parent := mt.StartSpan("http.request")
+	child := mt.StartSpan("db.query", tracer.ChildOf(parent.Context()))
+
+	assert.Len(t, mt.OpenSpans(), 2)
+	assert.Contains(t, mt.OpenSpans(), parent)
+	assert.Contains(t, mt.OpenSpans(), child)
+
+	child.Finish()
+	assert.Len(t, mt.OpenSpans(), 1)
+	assert.NotContains(t, mt.OpenSpans(), child)
 
+	parent.Finish()
+	assert.Empty(t, mt.OpenSpans())
+}
+
+func TestTracerReset(t *testing.T) {
 	assert := assert.New(t)
+	mt := newMockTracer()
+
+	span := mt.StartSpan("db.query")
+	assert.Len(mt.openSpans, 1)
+
+	span.Finish()
 	assert.Len(mt.finishedSpans, 1)
+	assert.Len(mt.openSpans, 0)
```

Comment:
It looks like `mt.openSpans` is already empty when `Reset()` is called. It would be good to ensure `Reset()` will clear any open spans too.